chat-widget/src/contexts/createReducer.ts

```diff
@@ -287,12 +288,21 @@ export const createReducer = () => {
                     ...action.payload as ILiveChatWidgetContext
                 };
 
-            case LiveChatWidgetActionType.SET_CONVERSATION_ENDED_BY_AGENT:
+            case LiveChatWidgetActionType.SET_CONVERSATION_ENDED_BY_AGENT_EVENT_RECEIVED:
                 return {
                     ...state,
                     appStates: {
                         ...state.appStates,
-                        conversationEndedByAgent: action.payload as boolean
+                        conversationEndedByAgentEventReceived: action.payload as boolean
+                    }
+                };
+                
+            case LiveChatWidgetActionType.SET_CONVERSATION_ENDED_BY:
```

Comment:
why do we need 2 different action types?  , when the agent set_conversation_ended_by ends the conversation should reflect that scenario, by that making SET_CONVERSATION_ENDED_BY_AGENT_EVENT_RECEIVED redundant.

Reply:
One reason pointed out below

Another reason is since conversationEndCallback logic is injected with states during the start of chat, it does not contain the latest state changes and can only contain logic to trigger certain events upon which we take action using the latest state changes. Hence when this event triggers we have no way of knowing if it is a valid scenario until we run some logic against it with the latest app states and set the correct behavior
So here I went with having conversationEndedByAgentEventReceived state flag raised whenever we receive an ended event from backend
Then conversationEndedBy state flag set when we want to take some conversationEnd action within our app based on if it is a valid scenario of agent ended chat